src/Sarif.Sarifer/BackgroundAnalyzerBase.cs

```diff
@@ -49,68 +39,95 @@ public abstract class BackgroundAnalyzerBase : IBackgroundAnalyzer
         /// <inheritdoc/>
         public abstract string ToolSemanticVersion { get; }
 
+        private const int DefaultBufferSize = 1024;
+
         /// <inheritdoc/>
-        public async Task AnalyzeAsync(string path, string text, CancellationToken cancellationToken)
+        public async Task<Stream> AnalyzeAsync(string path, string text, CancellationToken cancellationToken)
         {
             text = text ?? throw new ArgumentNullException(nameof(text));
 
             string solutionDirectory = await GetSolutionDirectoryAsync().ConfigureAwait(continueOnCapturedContext: false);
 
-            using (Stream stream = new MemoryStream())
-            using (TextWriter writer = new StreamWriter(stream, Encoding.UTF8))
+            // If we don't have a solutionDirectory, then, we don't need to analyze.
+            if (string.IsNullOrEmpty(solutionDirectory))
             {
-                using (SarifLogger sarifLogger = this.MakeSarifLogger(writer))
+                return Stream.Null;
+            }
+
+            var stream = new MemoryStream();
+            try
+            {
+                using (var writer = new StreamWriter(stream, Encoding.UTF8, DefaultBufferSize, leaveOpen: true))
                 {
-                    sarifLogger.AnalysisStarted();
+                    using (SarifLogger sarifLogger = this.MakeSarifLogger(writer))
+                    {
+                        sarifLogger.AnalysisStarted();
 
-                    // TODO: What do we do when path is null (text buffer with no backing file)?
-                    Uri uri = path != null ? new Uri(path, UriKind.Absolute) : null;
+                        // TODO: What do we do when path is null (text buffer with no backing file)?
+                        Uri uri = path != null ? new Uri(path, UriKind.Absolute) : null;
 
-                    this.AnalyzeCore(uri, text, solutionDirectory, sarifLogger, cancellationToken);
+                        this.AnalyzeCore(uri, text, solutionDirectory, sarifLogger, cancellationToken);
 
-                    sarifLogger.AnalysisStopped(RuntimeConditions.None);
-                }
+                        sarifLogger.AnalysisStopped(RuntimeConditions.None);
+                    }
 
-                await writer.FlushAsync().ConfigureAwait(continueOnCapturedContext: false);
+                    await writer.FlushAsync().ConfigureAwait(continueOnCapturedContext: false);
+                }
 
-                await this.WriteToSinksAsync(path, stream).ConfigureAwait(false);
+                return stream;
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception)
```

Comment:
>Exception [](start = 19, length = 9)

Why are we eating all exceptions here? We shouldn't do that, as CA1031 tells you. First, in cases where it is actually necessary for this kind of code, there should always be a logging mechanism capturing details. In this case, why would we catch this at all? The add-in is behaving poorly, well, let it crash. The danger of leaving this code as is is that analysis could be failing catastrophically and the user will never know. Meanwhile, analysis is completely disabled and problems could be sneaking into code.

Reply:
ok. removing this.

---
In reply to: [536220395](https://github.com/microsoft/sarif-visualstudio-extension/pull/310#discussion_r536220395) [](ancestors = 536220395)